octobot_evaluators/evaluators/abstract_evaluator.py

```diff
@@ -188,6 +267,14 @@ async def evaluation_completed(self,
             if eval_note is None:
                 eval_note = self.eval_note if self.eval_note is not None else common_constants.START_PENDING_EVAL_NOTE
 
+            if self.use_cache():
+                ctx = context or util.local_trading_context(self, symbol, time_frame, eval_time)
```

Comment:
todo: use cache object instead of a context

Reply:
Actually it's not that simple as context can also already exist and be given an as argument. After thinking this through, I think we should keep using a context object

Reply:
So we need to create a context to access cache? I think we have a code design issue here

Reply:
You are right, seen from this angle, there is an issue, I'll change this